macros/multiple_databases/timestamp_from_parts.sql

```diff
@@ -1,18 +1,37 @@
 {%- macro timestamp_from_parts(date_field, time_field) -%}
 
 {%- if target.type == 'snowflake' -%}
-    timestamp_from_parts({{ date_field }}, {{ time_field }})
+    case
+        when len({{ date_field }}) > 0 and len({{time_field}}) > 0
+            then timestamp_from_parts({{ date_field }}, {{ time_field }})
+        else
+            timestamp_from_parts(NULL, NULL)
+    end
 {%- elif target.type == 'sqlserver' -%}
-    datetime2fromparts(
-        datepart(year, {{ date_field }}),
-        datepart(month, {{ date_field }}),
-        datepart(day, {{ date_field }}),
-        datepart(hour, {{ time_field }}),
-        datepart(minute, {{ time_field }}),
-        datepart(second, {{ time_field }}),
-        datepart(millisecond, {{ time_field }}),
-        3
-    )
+    case
+        when len({{ date_field }}) > 0 and len({{time_field}}) > 0
+            then datetime2fromparts(
+            datepart(year, {{ date_field }}),
+            datepart(month, {{ date_field }}),
+            datepart(day, {{ date_field }}),
+            datepart(hour, {{ time_field }}),
+            datepart(minute, {{ time_field }}),
+            datepart(second, {{ time_field }}),
+            datepart(millisecond, {{ time_field }}),
+            3
+            )
+        else
+            datetime2fromparts(
+            datepart(year, NULL),
+            datepart(month, NULL),
+            datepart(day, NULL),
+            datepart(hour, NULL),
+            datepart(minute, NULL),
+            datepart(second, NULL),
+            datepart(millisecond, NULL),
```

Comment:
I think this can be simplified to `datetime2fromparts(NULL, NULL, NULL, NULL, NULL, NULL, NULL, 3)`. Please also look at the indenting.